### PR TITLE
Add formatting and Markdown sections to submitting.md

### PR DESCRIPTION
**Pull request by IlonaSilverwood**

### CODE REVIEW

File: docs/submitting.md [hunk 1]
```diff
@@ -106,6 +106,297 @@ As this short list shows, JOSS papers are only expected to contain a limited set
...
+    the North Sea. Sylt (Danish: *Slid*) is Germany's northernmost
+    island.](sylt.jpg){#sylt width="100%"}
+
+#### Tables and figures
+
+Tables and figures can be referenced if they are given a *label* in the caption. In pure Markdown, this can be done by adding an empty span `[]{label="floatlabel"}` to the caption. LaTeX syntax is supported as well: `\label{floatlabel}`.
+
+Link to a float element, i.e., a table or figure, with `\ref{identifier}` or `\autoref{identifier}`, where `identifier` must be defined in the float's caption. The former command results in just the float's number, while the latter inserts the type and number of the referenced float. E.g., in this document `\autoref{proglangs}` yields "\autoref{proglangs}", while `\ref{proglangs}` gives "\ref{proglangs}".
+
+: Comparison of programming languages used in the publishing tool. []{label="proglangs"}
```
Comment (tarleb) [inline docs/submitting.md:200]: This should probably be made part of the code, i.e., indented by 4 spaces.

File: docs/submitting.md [hunk 2]
```diff
@@ -106,6 +106,297 @@ As this short list shows, JOSS papers are only expected to contain a limited set
...
+    | Lua      | dynamic, strong | yes               | strict     | 1993    |
+    | C        | static, weak    | no                | strict     | 1972    |
+
+#### Equations
+
+Cross-references to equations work similarly to those for floating elements. The difference is that, since captions are not supported for equations, the label must be included in the equation:
+
+    $$a^n + b^n = c^n \label{fermat}$$
+
+Referencing, however, is identical, with `\autoref{eq:fermat}` resulting in "\autoref{eq:fermat}".
```
Comment (tarleb) [inline docs/submitting.md:214]: `"\autoref{eq:format}"` => "Equation 1".

File: docs/submitting.md [hunk 3]
```diff
@@ -106,6 +106,297 @@ As this short list shows, JOSS papers are only expected to contain a limited set
...
+
+$$a^n + b^n = c^n \label{eq:fermat}$$
```
Comment (tarleb) [inline docs/submitting.md:216]: I'm not sure how Sphinx handles this – maybe we should just remove that line.

**Review by tarleb — COMMENTED** [review]
LGTM!

There are some examples that relied on the document being processed with Inara, which won't work with Sphinx/read-the-docs. Those commands must be replaced with the expected text, and I've added inline comments wherever I found them.

**Comment by arfon** [comment]
Weird. It looks like these changes haven't been picked up by readthedocs yet?

**Comment by arfon** [comment]
Looks like the docs build failed with:

```
Running Sphinx v1.8.6
loading translations [en]... done
making output directory...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 9 source files that are out of date
updating environment: 9 added, 0 changed, 0 removed
reading sources... [ 11%] editing
/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/recommonmark/parser.py:65: UserWarning: Container node skipped: type=document
  warn("Container node skipped: type={0}".format(mdnode.t))
reading sources... [ 22%] editorial_bot
reading sources... [ 33%] index
reading sources... [ 44%] installing
reading sources... [ 55%] policies
reading sources... [ 66%] review_checklist
reading sources... [ 77%] review_criteria
reading sources... [ 88%] reviewer_guidelines
reading sources... [100%] submitting

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 11%] editing
writing output... [ 22%] editorial_bot
writing output... [ 33%] index
writing output... [ 44%] installing
writing output... [ 55%] policies
writing output... [ 66%] review_checklist
writing output... [ 77%] review_criteria
writing output... [ 88%] reviewer_guidelines
writing output... [100%] submitting

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/cmd/build.py", line 304, in build_main
    app.build(args.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/application.py", line 341, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 347, in build_update
    len(to_build))
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 412, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 593, in write
    self._write_serial(sorted(docnames))
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 601, in _write_serial
    doctree = self.env.get_and_resolve_doctree(docname, self)
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/environment/__init__.py", line 627, in get_and_resolve_doctree
    self.apply_post_transforms(doctree, docname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/environment/__init__.py", line 674, in apply_post_transforms
    transformer.apply_transforms()
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/transforms/__init__.py", line 90, in apply_transforms
    Transformer.apply_transforms(self)
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/docutils/transforms/__init__.py", line 171, in apply_transforms
    transform.apply(**kwargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/sphinx/transforms/post_transforms/compat.py", line 51, in apply
    equation = node['latex']
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/docutils/nodes.py", line 652, in __getitem__
    return self.attributes[key]
KeyError: 'latex'

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/joss/envs/latest/lib/python2.7/site-packages/docutils/nodes.py", line 652, in __getitem__
    return self.attributes[key]
KeyError: 'latex'
The full traceback has been saved in /tmp/sphinx-err-DmqQFK.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

@tarleb @IlonaSilverwood – could you investigate please?

**Pull request merged**